src/search/search-executor/search-enrichers.ts

```diff
@@ -126,6 +170,47 @@ export const Enrichments = {
 		},
 		mapper: ({ linkedIds }) => linkedIds
 	} satisfies EnrichmentData<LastCallContent, Omit<LinkToLastCall, 'type'>[]>,
+	[Enrichment.CfgInformation]: {
+		enrichElement: (e, search, _data, _args, prev) => {
+			const searchContent: CfgInformationSearchContent = search.enrichmentContent(Enrichment.CfgInformation);
+			return {
+				...prev,
+				isRoot:      searchContent.cfg.graph.rootIds().has(e.node.info.id),
+				isReachable: searchContent.reachableNodes?.has(e.node.info.id)
+			};
+		},
+		enrichSearch: (_search, data, args, prev) => {
+			args = {
+				forceRefresh:         false,
+				checkReachable:       true,
+				simplificationPasses: [...DefaultCfgSimplificationOrder, 'analyze-dead-code'],
+				...args
+			};
+
+			// short-circuit if we already have a cfg stored
+			if(!args.forceRefresh && prev?.simpleCfg) {
+				return prev;
+			}
+
+			const content: CfgInformationSearchContent = {
+				...prev,
+				cfg: extractCfg(data.normalize, data.config, data.dataflow.graph, args.simplificationPasses),
+			};
+			if(args.checkReachable) {
+				const reachable = new Set<NodeId>();
+				visitCfgInOrder(content.cfg.graph, content.cfg.entryPoints, node => {
```

Comment:
Please outsource into a general purpose function this should be used in the dead code removal